infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -82,8 +88,6 @@
     // a new instance of this class should be created after deleting
     // the files of the old instance
     private LogMetadata logMetadata;
-    private long lastSegment;
-    private volatile long startingAddress;
```

Comment:
What is the advantage of removing this?
Now all methods accessing these fields need to go via the datastore. We might pay the cost of IO calls.
Even if we argue that this is cached, the get is synchronized. This lock is shared among all components.
This can cause a performance impact. We have seent the datastore being a performance bottleneck in the past.

Reply:
It can't cause a performance issue, because we use it only when we create a streamLog, trim it or release it. It happens pretty rear so it just can't affect performance. You can also see that we read and save the variable same number of times in prev version and the current version. I don't see any reasons to overcomplicate logic and have those variables. 

The advantage is - it comes more consistent. You use only your datastore to get and update those variables, it's synchronized, it guarantees ordering and so on.  Also in prev version, you could have the variable is not equals to the value in the datastore, in current version we improved that.

Reply:
even rare might have effects on the performance though.

Reply:
Trimming happens every... how long? 15 minutes. It just can't

Reply:
I don't agree with this.
This call would be made in `isTrimmed()`. This is called on every operation - read and append.

1. Do not introduce performance or behaviour changes in a cleanup PR.
2. If you still want to change this then create the same volatile variables in the StreamLogDatastore and cache them and update them there.

As I said before, since the synchronized lock is shared by all servers, there can be contention on the lock causing performance degradation.

Reply:
oh, you are right, will return startingAddress variable then

Reply:
startingAddress and tailSegment moved to StreamLogDataStore, added a a test